ACE-update.pl

```diff
@@ -0,0 +1,549 @@
+
+# Usuall, $dir ends up being /proj/sot/acis/FLU-MON/
+if($isBackup) {
+    $dir=GetFluMon("colossus-v");
+}
+else{
+    $dir=${xcanuck_flu}
+}
+$old_umask=umask(002);
+$umask=002;
+
+#Key is the mission history file, value is the load history file
+%history_files=("FPHIST-2001","ACIS-FPHIST",
+		"GRATHIST-2001","ACIS-GRATHIST",
+		"OBSHIST","ACIS-OBSHIST",
+		"TLMHIST","ACIS-TLMHIST",
+		"TSCHIST","ACIS-TSCHIST",
+		"DITHHIST","ACIS-DITHHIST",
+                "SCS155HIST", "ACIS-SCS155HIST");
+
+
+   
+#
+#----------------------------------------
+#Getoptions: looking for test directory
+#----------------------------------------
+GetOptions('test=s',\$test_dir);
+if($test_dir)
+  {
+    print "\n\nACE-update - Running ACE-update in TEST MODE.\n\n";
+    $test=1;
+    $dir=$test_dir;
+    $load_file = "$dir/current_load.txt";
+  } 
+else
+  {
+    $load_file = "$base_dir/LoadReviews/current_load.txt";
+    $test = 0;
+  }
+
+#Collect the week and year information
+open(LOAD, $load_file) || die 
+    "Could not open $load_file for input\n";
+while(<LOAD>)
+{
+    chop;
+    @info=split /\//;
+    $load=$_;
+}
+$week=@info[5];
+$year=@info[4];
+
+#Set up a read loop to get the right path
+  $ans='N';# default for loop 
+    
+    while($ans !~ /[Yy]/)
+{
+    print "$load\n";
+    print STDOUT "Is this the right path to approved load? (y or n) ";
+    chop($ans=<STDIN>); 
+    
+    if ($ans !~ /[Yy]/)
+    {
+	print STDOUT "Please provide path:";
+	chop($load=<STDIN>);
+	@info=split(/\//,$load);
+	$week=@info[5];
+	$year=@info[4];
+     }
+} #end while loop
+
+#Need to set the remainder of the file paths here to capture the year.
+
+$load_dir=$load;
+    $dir2="$base_dir/LoadReviews/$year";
+$dir3="/data/acis33/LoadReviews/$year";
+
+&confirm_cat(%history_files);
+#------------------------------------------------
+# Loop through each history file.
+#------------------------------------------------
+while (($key, $value) = each(%history_files)){
+    print "...Updating history file $key.dat...\n";
+    print "Copying $dir/$key.dat to $dir/$key-temp.dat\n"; 
+    copy("$dir/$key.dat","$dir/$key-temp.dat");
+    #print "Read for internal Cat\n";
+    #----------------------------------------
+    #combine the files
+    #----------------------------------------
+    &internal_cat("$dir/$key-temp.dat","$load_dir/$value.dat",
+		  "$dir/$key.dat"); 
+}
+
+#--------------------------------------------------
+# Now that files are all updated, then put the copy
+# in a separate loop
+#--------------------------------------------------
+$hist_fls_str = "";
+
+while (($key, $value) = each(%history_files)){
+    $hist_fls_str .= "$dir/${key}.dat ";
+   #----------------------------------------
+    #Change the group to acisops
+    #----------------------------------------
+    &chgrp_by_name("acisops","$dir/$key.dat");
+}
+
+$HEAD_flu = "han-v:$xcanuck_flu"; # Global history file directory
```

Comment:
Looks like you're using han-v for the global; is that still how you want to do it, or should han-v be excised altogether?